### ISSUE TITLE
PyAsn1Error: Uninitialized ASN.1 value upon `who_am_i` with pyasn1==0.3.1

### ISSUE THREAD

**Issue by sgratzl** [cannatag/ldap3#388]
ldap3 version: 2.2.3
pyasn1 version: 0.3.1 (latest)

## Observed:

after the latest update of the pyasn1 library I get an exception when executing the `who_am_i()` standard action. 

see log:
```
2017-07-27T06:48:54.100639447Z     my_username = connection.extend.standard.who_am_i()
2017-07-27T06:48:54.100647160Z   File "/usr/local/lib/python2.7/site-packages/ldap3/extend/__init__.py", line 62, in who_am_i
2017-07-27T06:48:54.100655523Z     controls).send()
2017-07-27T06:48:54.100662710Z   File "/usr/local/lib/python2.7/site-packages/ldap3/extend/operation.py", line 53, in send
2017-07-27T06:48:54.100670510Z     resp = self.connection.extended(self.request_name, self.request_value, self.controls)
2017-07-27T06:48:54.100678297Z   File "/usr/local/lib/python2.7/site-packages/ldap3/core/connection.py", line 1009, in extended
2017-07-27T06:48:54.100686180Z     response = self.post_send_single_response(self.send('extendedReq', request, controls))
2017-07-27T06:48:54.100694237Z   File "/usr/local/lib/python2.7/site-packages/ldap3/strategy/base.py", line 291, in send
2017-07-27T06:48:54.100701973Z     self.connection.request = BaseStrategy.decode_request(ldap_message)
2017-07-27T06:48:54.100709137Z   File "/usr/local/lib/python2.7/site-packages/ldap3/strategy/base.py", line 592, in decode_request
2017-07-27T06:48:54.100716733Z     result = extended_request_to_dict(component)
2017-07-27T06:48:54.100723796Z   File "/usr/local/lib/python2.7/site-packages/ldap3/operation/extended.py", line 58, in extended_request_to_dict
2017-07-27T06:48:54.100731563Z     return {'name': str(request['requestName']), 'value': str(request['requestValue']) if request['requestValue'] else None}
2017-07-27T06:48:54.100742190Z   File "/usr/local/lib/python2.7/site-packages/pyasn1/type/base.py", line 275, in __nonzero__
2017-07-27T06:48:54.100755283Z     return self._value and True or False
2017-07-27T06:48:54.100767223Z   File "/usr/local/lib/python2.7/site-packages/pyasn1/type/base.py", line 191, in plug
2017-07-27T06:48:54.100781130Z     raise error.PyAsn1Error('Uninitialized ASN.1 value ("%s" attribute looked up)' % name)
2017-07-27T06:48:54.100803123Z PyAsn1Error: Uninitialized ASN.1 value ("__nonzero__" attribute looked up) 
```

## Workaround
pin `pyasn1==0.2.3`

see also https://github.com/etingof/pyasn1/issues/52

**Comment by etingof** [comment]
This [fix](https://github.com/cannatag/ldap3/pull/387) would likely help.

Let me know if it does not so I could address that!

**Comment by cannatag** [comment]
Fixed in release 2.3 of ldap3.
